src/java/org/apache/cassandra/db/filter/RowFilter.java

```diff
@@ -383,6 +383,30 @@ public RowFilter without(ColumnMetadata column, Operator op, ByteBuffer value)
         return withNewExpressions(newExpressions);
     }
 
+    /**
+     * @return a copy of this filter without any primary key expressions, or simply itself if non exist already
+     */
+    public RowFilter withoutKeyExpressions()
```

Comment:
I don't think this is the right approach to fix BETWEEN#isSatified issue.. This does not fix root cause

Reply:
Talked to Caleb, it makes sense to skip RFP if only primary key columns are filtered

Reply:
BETWEEN itself now fixed via [CASSANDRA-19878](https://github.com/blerer/cassandra/tree/CASSANDRA-19878)